File: app/src/main/java/com/lofts/dictionary/adapter/SampleAdapter.java
package com.lofts.dictionary.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.lofts.dictionary.R;
import com.lofts.dictionary.bean.Sample;

import java.util.ArrayList;
import java.util.List;

public class SampleAdapter extends BaseAdapter {

    private Context mContext;
    private List<Sample> mListSample;
    private OnPlaySampleListener mListener;

    public void setPlaySampleListener(OnPlaySampleListener listener) {
        mListener = listener;
    }

    public SampleAdapter(Context context) {
        mContext = context;
        mListSample = new ArrayList<>();
    }

    public void resetList(List<Sample> list) {
        mListSample.clear();
        mListSample.addAll(list);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return mListSample.size();
    }

    @Override
    public Object getItem(int position) {
        return mListSample.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        convertView = LayoutInflater.from(mContext).inflate(R.layout.item_sample, parent, false);

        TextView english = convertView.findViewById(R.id.textview_english);
        TextView chinese = convertView.findViewById(R.id.textview_chinese);
        ImageView speak = convertView.findViewById(R.id.imageview_speak);

        Sample sample = mListSample.get(position);
        english.setText(sample.getEng());
        chinese.setText(sample.getChn());
        speak.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mListener.playSample(sample.getMp3Url());
            }
        });

        return convertView;
    }


    public interface OnPlaySampleListener {
        void playSample(String url);
    }
}

File: app/src/main/java/com/lofts/dictionary/adapter/SampleAdapter.kt
package com.lofts.dictionary.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.lofts.dictionary.R
import com.lofts.dictionary.bean.Sample

/**
 * @name      DictionaryKotlin
 * @package   com.lofts.dictionary.adapter
 * @author    Administrator
 * @time      2018/12/19 17:57
 * @describe
 */

class SampleAdapter(context: Context) : BaseAdapter() {

    private var mContext: Context
    private var mListSample: MutableList<Sample>
    private var mListener: OnPlaySampleListener? = null

    init {
        mContext = context
        mListSample = ArrayList()
    }

    fun setPlaySampleListener(listener: OnPlaySampleListener) {
        mListener = listener
    }

    fun resetList(list: List<Sample>?) {
        mListSample.clear()
        mListSample.addAll(list!!)
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var convertView = LayoutInflater.from(mContext).inflate(R.layout.item_sample, parent, false)

        var english = convertView.findViewById(R.id.textview_english) as TextView
        var chinese = convertView.findViewById(R.id.textview_chinese) as TextView
        var speak = convertView.findViewById(R.id.imageview_speak) as ImageView

        var sample = mListSample[position]
        english.text = sample.eng
        chinese.text = sample.chn

        speak.setOnClickListener {
            mListener?.playSample(sample.mp3Url)
        }

        return convertView
    }

    override fun getItem(position: Int): Any {
        return mListSample[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return mListSample.size
    }

    interface OnPlaySampleListener {
        fun playSample(url: String)
    }

}